common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -230,6 +240,70 @@ protected final void setAuthorizationEndpoint(final String authorizationEndpoint
         mAuthorizationEndpoint = authorizationEndpoint;
     }
 
+    public AuthorizationResult getDeviceCode(@NonNull final MicrosoftStsAuthorizationRequest authorizationRequest) throws IOException {
+        final String methodName = ":getDeviceCode";
+
+        final String urlBody = ((MicrosoftStsOAuth2Configuration) mConfig).getAuthorityUrl().toString() + "/oauth2/v2.0/devicecode";
```

Comment:
Can we extract this into MicrosoftStsOAuth2Configuration.getDeviceCodeEndpoint()? 

No need to fetch the endpoint from .well-known/openid-configuration yet. (If you could add a TODO there that we're pending server change, that would be really great.)

Reply:
Ok, will do!